osgp/shared/osgp-dto/src/main/java/org/opensmartgridplatform/dto/valueobjects/smartmetering/DeCoupleMbusDeviceResponseDto.java

```diff
@@ -11,26 +11,29 @@
 
     private static final long serialVersionUID = -4454979905929290745L;
 
-    private final Short channel;
-    private final String mBusDeviceIdentification;
+    private String mbusDeviceIdentification;
+    private final ChannelElementValuesDto channelElementValues;
 
-    public DeCoupleMbusDeviceResponseDto(final String mBusDeviceIdentification, final Short channel) {
-        this.mBusDeviceIdentification = mBusDeviceIdentification;
-        this.channel = channel;
+    public DeCoupleMbusDeviceResponseDto(final ChannelElementValuesDto channelElementValues) {
+        super("Decouple Mbus Device was successful");
+        this.channelElementValues = channelElementValues;
     }
 
     @Override
     public String toString() {
-        return "DeCoupleMbusDeviceResponseDto [channel=" + this.channel + ", mBusDeviceIdentification="
-                + this.mBusDeviceIdentification + "]";
+        return "DeCoupleMbusDeviceResponseDto [channel=" + this.channelElementValues.getChannel()
+                + ", mbusDeviceIdentification=" + this.mbusDeviceIdentification + "]";
     }
 
-    public Short getChannel() {
-        return this.channel;
+    public ChannelElementValuesDto getChannelElementValues() {
+        return this.channelElementValues;
     }
 
-    public String getmBusDeviceIdentification() {
-        return this.mBusDeviceIdentification;
+    public String getMbusDeviceIdentification() {
+        return this.mbusDeviceIdentification;
     }
 
+    public void setMbusDeviceIdentification(final String mbusDeviceIdentification) {
```

Comment:
I would keep to an immutable object here. Use lombok and add mbusDeviceIdentification to constructor.

Reply:
mbusDeviceIdentification not available at the moment of creation of the dto object